### DISCUSSION TITLE
v2021.11.9

### DISCUSSION THREAD

**Discussion by ScratMan** [ScratMan/HASmartThermostat#30] [Announcements]
# Breaking change !!
## What's new:
* **Add optional `unique_id` parameter:**  _Breaking change !_
Closes #22.
It allows to customize the entity through UI. Please note that leaving this setting empty may duplicate the previously existing entity, and the old one will then be marked as unavailable.
Example:
  * Old entity becoming unavailable : `climate.room_thermostat`
  * New entity : `climate.room_thermostat_2`
You should note the ID of the old entity, then delete the old entity in the UI and rename the new entity using the same ID as old entity to restore functionality of all your automations, scripts, cards, etc...

To avoid this, update Smart Thermostat using HACS, then stop Home Assistant, set the unique_id parameter in YAML with the ID of the old smart thermostat entity as shown in Developers tools > States (example if old entity ID is `climate.room_thermostat` in developers tools, then write `unique_id: room_thermostat` in the YAML file), save file and then start Home Assistant.

* **Add service to set PID gains :**
Use the `smart_thermostat.set_pid_gain` service to adjust the PID gains without requiring a restart of Home Assistant. Values are saved to Home Assistant database and restored after a restart. Please consider saving the final gain parameters in YAML configuration file when satisfied to keep it safe in case of database corruption.

* **Add service to set preset modes temperatures:**
Use the `smart_thermostat.set_preset_temp` service to set the temperatures for the preset modes. Closes #17.

* **Add service to clear the integral part:**
Use the `smart_thermostat.clear_integral` service to reset the integral part of the PID controller to 0. Useful when tuning the PID gains to quickly test the behavior without waiting the integral to stabilize by itself.

* **Add integration reload service:**
Use the `smart_thermostat.reload` service to reload the thermostat from scratch without requiring a restart of Home Assistant.

## Bug fixes:
* Fix integral windup when set point is changed while heating is on and modulating.
* Fix logging issue when autotune is successful.
* Fix error when calling service `climate.set_preset_mode` to set preset to 'none' while no preset is active (#28)

**Full Changelog**: https://github.com/ScratMan/HASmartThermostat/compare/2021.11.8...2021.11.9

<hr /><em>This discussion was created from the release <a href='https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.11.9'>v2021.11.9</a>.</em>

**Comment by Stefano0042** [comment]
Hello Adrien, my personal congrats for this good job, a very useful component! Thank you again.
